huaweicloud/services/dds/resource_huaweicloud_dds_lts_log.go

```diff
@@ -0,0 +1,338 @@
+	}
+
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", utils.PathSearch("id", instance, nil)),
+		d.Set("log_type", utils.PathSearch("log_type", ltsConfigs[0], nil)),
+		d.Set("lts_group_id", utils.PathSearch("lts_group_id", ltsConfigs[0], nil)),
+		d.Set("lts_stream_id", utils.PathSearch("lts_stream_id", ltsConfigs[0], nil)),
+		d.Set("enabled", utils.PathSearch("enabled", ltsConfigs[0], nil)),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func buildDeleteDdsLtsLogBodyParams(d *schema.ResourceData) map[string]interface{} {
+	ltsConfigs := map[string]interface{}{
+		"lts_configs": []map[string]interface{}{
+			{
+				"instance_id": d.Get("instance_id"),
+				"log_type":    d.Get("log_type"),
+			},
+		},
+	}
+	return ltsConfigs
+}
+
+func resourceDdsLtsLogDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var (
+		deleteDdsLtsLogHttpUrl = "v3/{project_id}/instances/logs/lts-configs"
+		deleteDdsLtsLogProduct = "dds"
+	)
+	deleteDdsLtsLogClient, err := cfg.NewServiceClient(deleteDdsLtsLogProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DDS Client: %s", err)
+	}
+
+	deleteDdsLtsLogPath := deleteDdsLtsLogClient.Endpoint + deleteDdsLtsLogHttpUrl
+	deleteDdsLtsLogPath = strings.ReplaceAll(deleteDdsLtsLogPath, "{project_id}", deleteDdsLtsLogClient.ProjectID)
+
+	deleteDdsLtsLogOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			202,
+		},
+	}
+	deleteDdsLtsLogOpt.JSONBody = utils.RemoveNil(buildDeleteDdsLtsLogBodyParams(d))
+
+	retryFunc := func() (interface{}, bool, error) {
+		deleteDdsLtsLogResp, err := deleteDdsLtsLogClient.Request("DELETE", deleteDdsLtsLogPath, &deleteDdsLtsLogOpt)
+		retry, err := handleMultiOperationsError(err)
+		return deleteDdsLtsLogResp, retry, err
+	}
+	_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     ddsInstanceStateRefreshFunc(deleteDdsLtsLogClient, d.Id()),
+		WaitTarget:   []string{"normal"},
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		DelayTimeout: 10 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return diag.Errorf("error unassociating DDS with LTS log: %s", err)
+	}
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"true"},
+		Target:     []string{"false"},
+		Refresh:    ddsLtsConfigRefreshFunc(deleteDdsLtsLogClient, d.Id()),
+		Timeout:    d.Timeout(schema.TimeoutDelete),
+		Delay:      1 * time.Second,
+		MinTimeout: 2 * time.Second,
+	}
+
+	_, err = stateConf.WaitForStateContext(ctx)
+	if err != nil {
+		return diag.Errorf("error waiting for unassociating DDS (%s) with LTS log to become ready: %s ", d.Id(), err)
+	}
+
+	return resourceDdsLtsLogRead(ctx, d, meta)
+}
+
+func filterInstanceLtsConfig(instanceID string, resp interface{}) interface{} {
```

Comment:
you can rename this func to `FilterInstanceLtsConfig` and use it in the test

Reply:
done